pgpy/types.py

```diff
@@ -59,7 +59,7 @@ class Armorable(six.with_metaclass(abc.ABCMeta)):
     __armor_regex = re.compile(r"""# This capture group is optional because it will only be present in signed cleartext messages
                          (^-{5}BEGIN\ PGP\ SIGNED\ MESSAGE-{5}(?:\r?\n)
                           (Hash:\ (?P<hashes>[A-Za-z0-9\-,]+)(?:\r?\n){2})?
-                          (?P<cleartext>(.*\r?\n)+?)(?:\r?\n)?
+                          (?P<cleartext>(\r?\n|.)+?)(?:\r?\n)?
```

Comment:
Ah, yep - that's why thorough testing is important, whoops 8). I came up with a couple of other working expressions and profiled them alongside yours and found that `(?P<cleartext>(.*\r?\n)*(.*(?=\r?\n-{5})))(?:\r?\n)` works and is also a bit faster